### PR TITLE
Add new utility functions: m/explain-data, m/data-explainer, mu/keys

### PR DESCRIPTION
**Pull request by opqdonut**
The main issue I'm trying to fix here is not being able to serialize the output of m/explain. See commit message for more.

mu/keys was just something simple I've been missing so I'm adding it in the same PR :)

### CODE REVIEW

File: src/malli/util.cljc [hunk 1]
```diff
@@ -257,6 +257,12 @@
    (let [schema (m/schema (or ?schema :map) options)]
      (when schema (m/-get schema [::m/find k] nil)))))
 
+(defn keys
+  "Like [[clojure.core/keys]], but for EntrySchemas."
+  [?schema]
```
Comment (opqdonut) [inline src/malli/util.cljc:262]: I'm not sure what the `?schema` convention was for, but I've followed it here.

File: src/malli/core.cljc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 (ns malli.core
   (:refer-clojure :exclude [eval type -deref deref -lookup -key])
   #?(:cljs (:require-macros malli.core))
-  (:require #?(:clj [clojure.walk :as walk])
+  (:require [clojure.walk :as walk]
```
Comment (opqdonut) [inline src/malli/core.cljc:4]: I'm not sure why clojure.walk was only required in clj mode. Perhaps to reduce cljs bundle size? If so, I can move the data-explainer to malli.util, or alternatively rewrite it without using clojure.walk.

**Review by ikitommi — CHANGES_REQUESTED** [review]
Good and clean PR, some changes requested.

File: src/malli/core.cljc [hunk 3]
```diff
@@ -2070,6 +2070,23 @@
...
+   (let [explainer' (explainer ?schema options)
+         schema->data (fn [s]
+                        (if (schema? s)
+                          (form s)
+                          s))]
+     (fn data-explainer
+       ([value]
+        (data-explainer value [] []))
+       ([value in acc]
+        (walk/prewalk schema->data (explainer' value in acc)))))))
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply src/malli/core.cljc:2088]: Good point. I was under the impression that the errors can somehow nest and you can have a `:schema` at any level, but what you say makes sense. Done.

File: test/malli/core_test.cljc [hunk 4]
```diff
@@ -2663,3 +2664,25 @@
 (deftest cat-catn-unparse-test
   (is (= ::m/invalid (m/unparse [:cat string? int? string?] [1 2 3])))
   (is (= ::m/invalid (m/unparse [:catn [:a string?] [:b int?] [:c string?]] {:a 1 :b 2 :c 3}))))
+
+(#:clj
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply test/malli/core_test.cljc]: added one

File: src/malli/core.cljc (hunk 2 above)
Comment (opqdonut) [reply src/malli/core.cljc:4]: good point, moved to malli.util

File: test/malli/util_test.cljc [hunk 5]
```diff
@@ -946,6 +946,13 @@
          (-> [:map [:a [:map [:b {:optional true} Int]]]]
              (mu/get-in [:a [::m/find :b]])))))
 
+(deftest keys-test
```
[the remark this answers is not in the document]
Comment (opqdonut) [reply test/malli/util_test.cljc:949]: Done!

**Comment by opqdonut** [comment]
I didn't rebase/squash anything to make reviewing easier. I can squash this into two nice commits if you want.

**Review by ikitommi — APPROVED** [review]
looks really good, please update the docs (at least CHANGELOG) before/after merging.

File: src/malli/util.cljc [hunk 6]
```diff
@@ -182,6 +182,35 @@
...
     @state))
 
+(defn data-explainer
+  "Like `m/explainer` but output is pure clojure data. Schema objects have been replaced with their m/form.
+   Useful when you need to serialise errrors."
+  ([?schema]
+   (data-explainer ?schema nil))
+  ([?schema options]
+   (let [explainer' (m/explainer ?schema options)
+         schema->data (fn [s]
```
Comment (ikitommi) [inline src/malli/util.cljc:192]: side-note: it's always a `Schema`, so `schema->data` could be replaced just with `m/form`.

**Pull request merged**